services/api/service.go

```diff
@@ -1335,9 +1333,47 @@ func (api *RelayAPI) handleSubmitNewBlock(w http.ResponseWriter, req *http.Reque
 		}
 	}()
 
+	// Get the current (in-memory) top bid value
+	api.builderTopBidValueLock.RLock()
+	localTopBidValue := api.builderTopBidValue[payload.ParentHash()]
+	api.builderTopBidValueLock.RUnlock()
+
+	// Without cancellations, discard lower or similar value submissions to previous top bid
+	if !isCancellationEnabled {
+		// Check the local top bid value first
+		if localTopBidValue != nil && payload.Value().Cmp(localTopBidValue) < 1 {
```

Comment:
This check should take into account the parent & cl fields (randao in particular)